main.go

```diff
@@ -80,41 +90,55 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
-	kubeConfig := ctrl.GetConfigOrDie()
-	kubeConfig.QPS = float32(qps)
-	kubeConfig.Burst = burst
-
 	if err := utilfeature.DefaultMutableFeatureGate.Set(featureGates); err != nil {
 		setupLog.Error(err, "Unable to set flag gates for known features")
 		os.Exit(1)
 	}
 
 	metrics.Register()
 
-	mgr, err := ctrl.NewManager(kubeConfig, ctrl.Options{
-		Scheme: scheme,
-		Metrics: server.Options{
-			BindAddress: metricsAddr,
-		},
-		WebhookServer: webhook.NewServer(
-			webhook.Options{
-				Port: 9443,
-			}),
-		HealthProbeBindAddress: probeAddr,
-		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "6d4f6a47.x-k8s.io",
-		// LeaderElectionReleaseOnCancel defines if the leader should step down voluntarily
-		// when the Manager ends. This requires the binary to immediately end when the
-		// Manager is stopped, otherwise, this setting is unsafe. Setting this significantly
-		// speeds up voluntary leader transitions as the new leader don't have to wait
-		// LeaseDuration time first.
-		//
-		// In the default scaffold provided, the program ends immediately after
-		// the manager stops, so would be fine to enable this option. However,
-		// if you are doing or is intended to do any operation such as perform cleanups
-		// after the manager stops then its usage might be unsafe.
-		// LeaderElectionReleaseOnCancel: true,
-	})
+	var options manager.Options
+
+	kubeConfig := ctrl.GetConfigOrDie()
+	if configFile == "" {
+		options = ctrl.Options{
```

Comment:
Why don't we get the defaults as defined in SetDefaults_Configuration and then override the couple of parameters that have corresponding cmd flags if set? This will unify all defaulting in one place.

Reply:
`config.Load` handles the case where the configFile is not set, so we just need to invoke it instead of defining defaults again here.

Reply:
Modify as you said, now check each flag individually and set the value.